### ISSUE TITLE
Schemer: no need to migrate schema from pod which hasn't been created

### ISSUE THREAD

**Issue by slimtom95** [Altinity/clickhouse-operator#479]
In a concile process, it's no need to migrate schema from the hosts that we know it's not exist and will be created later;

Schemer can really hang there for some time to wait invisible nodes respond.

**Comment by alex-zaitsev** [comment]
Schemer uses the load balancer and skip_unavailable_shards setting, so it should not go to inactive pods.

**Comment by SalleryWang** [comment]
the process spent too a lot  time，i think when we create new cluster，no need to migrating schema objects to host
![image](https://user-images.githubusercontent.com/45587738/88611350-6af31a00-d0bb-11ea-8c05-3fd3b73f3c8f.png)

**Comment by SalleryWang** [comment]
flag that mean the old chi is nil.

**Comment by SalleryWang** [comment]
@alex-zaitsev

**Comment by slimtom95** [comment]
Please read this log screenshots, it did ask invisible pods to answer. 

![image](https://user-images.githubusercontent.com/20624187/89004996-ff68b100-d335-11ea-9099-77c7a9bc92db.png)

**Comment by alex-zaitsev** [comment]
ok. For some reason it does not fail fast in your environment and waits 10 seconds timeout every time.

**Comment by Slach** [comment]
@SalleryWang @slimtom95 , could you answer to following questions:
- according to your comments with code https://github.com/Altinity/clickhouse-operator/issues/479#issuecomment-664735922
do you use a custom fork for `clickhouse-operator`?

- according to shared PDF, do you use clickhouse cluster with 32 shards with 3 replicas inside each shard?

look like your fork contains error in schemer.go and you tried get schema objects from unavailable hosts which not exists now

for example from shared PDF file

```
I0727 01:21:53.570763 1 schemer.go:247] Migrating schema objects to host 28-2
I0727 01:21:53.570839 1 schemer.go:193] Extracting replicated table definitions from [chi-ck-bo7s4lw7ax-31-0.ck￾bo7s4lw7ax.svc.cluster.local chi-ck-bo7s4lw7ax-31-1.ck￾bo7s4lw7ax.svc.cluster.local]
I0727 01:21:53.623319 1 connection.go:95] FAILED Query(http://
***:***@chi-ck-bo7s4lw7ax-0-0.ck-bo7s4lw7ax.svc.cluster.local:8123/) 
clickhouse: Code: 90, e.displayText() = DB::Exception: Empty list of 
columns passed (version 19.17.4.11 (official build))
 for SQL: SELECT DISTINCT 
database AS name, 
concat('CREATE DATABASE IF NOT EXISTS "', name, '"') AS 
create_db_query
FROM remote('chi-ck-bo7s4lw7ax-31-0.ckbo7s4lw7ax.svc.cluster.local,chi-ck-bo7s4lw7ax-31-1.ck￾bo7s4lw7ax.svc.cluster.local', system, tables)
WHERE database != 'system'
SETTINGS skip_unavailable_shards = 1

```

I don't know why your custom operator fork tried to connect to `chi-ck-bo7s4lw7ax-0-0` instead of `chi-ck-bo7s4lw7ax-28-2`?

**Comment by SalleryWang** [comment]
not a fork，it‘s origin

**Comment by Slach** [comment]
@SalleryWang which version?

**Comment by SalleryWang** [comment]
@Slach v0.10.0

**Comment by Slach** [comment]
@SalleryWang in v0.10.0
part of code which you share in https://github.com/Altinity/clickhouse-operator/issues/479#issuecomment-664735922

looks different than part of `worker.go` in current 0.10.0 
https://github.com/Altinity/clickhouse-operator/blob/master/pkg/controller/chi/worker.go#L398

![image](https://user-images.githubusercontent.com/105560/89144515-733dd000-d567-11ea-95c3-c2b82e95ad39.png)

ok. after internal discussion we decided to improve `schemer.go` to avoid create tables when creating ClickHouse cluster from scratch

**Comment by SalleryWang** [comment]
fine

**Comment by SalleryWang** [comment]
I just made a little change to avoid migrating schema from pod which hasn't been created, The premise is i use cloud disk. so I think this is fine

**Comment by SalleryWang** [comment]
@Slach

**Comment by alex-zaitsev** [comment]
Fixed in 0.11.0

**Comment by alex-zaitsev** [comment]
https://github.com/Altinity/clickhouse-operator/releases/tag/0.11.0
